types.go

```diff
@@ -121,6 +121,15 @@ type functionInfo struct {
 	statuser         Statuser
 }
 
+func (f functionInfo) canEvaluate(grad, hess bool) error {
```

Comment:
Could you add: "TODO(btracey): Think about making this an exported function when the constraint interface is designed"
It would be preferable to allow users to check without calling Local, but until we have the full story it may as well be private.